src/sentry/issues/escalating.py

```diff
@@ -178,3 +189,53 @@ def _extract_project_and_group_ids(groups: Sequence[Group]) -> Dict[int, List[in
         group_ids_by_project[group.project_id].append(group.id)
 
     return group_ids_by_project
+
+
+def get_group_daily_count(project_id: int, group_id: int) -> int:
+    """Return the number of events a group has had today"""
+    date_now = datetime.now().date()
+    start_date = date_now
+    end_date = date_now + timedelta(days=1)
+    query = Query(
+        match=Entity(EntityKey.Events.value),
+        select=[
+            Function("count", []),
+        ],
+        where=[
+            Condition(Column("project_id"), Op.EQ, project_id),
+            Condition(Column("group_id"), Op.EQ, group_id),
+            Condition(Column("timestamp"), Op.GTE, start_date),
+            Condition(Column("timestamp"), Op.LT, end_date),
+        ],
+    )
+    request = Request(dataset=Dataset.Events.value, app_id=IS_ESCALATING_REFERRER, query=query)
+    return int(raw_snql_query(request, referrer=IS_ESCALATING_REFERRER)["data"][0]["count()"])
+
+
+def is_escalating(group: Group) -> bool:
+    """Return boolean depending on if the group is escalating or not"""
+    forecast_cache_key = f"escalating-forecast:{group.id}"
+    escalating_forecast = cache.get(forecast_cache_key)
+    date_now = datetime.now().date()
+    if escalating_forecast is None:
+        escalating_forecast = EscalatingGroupForecast.fetch(group.project.id, group.id)
+        escalating_forecast = (
+            escalating_forecast.date_added,
+            escalating_forecast.forecast,
+        )
+
+        # Set the cache to be valid until the next weekly escalating forecast task is run
+        forecast_cache_duration = (
+            (escalating_forecast[0] + timedelta(days=ONE_WEEK_DURATION)).date() - date_now
+        ).total_seconds()
+        cache.set(forecast_cache_key, escalating_forecast, forecast_cache_duration)
+
+    # Check if current event occurance is greater than forecast for today's date
+    group_daily_count = get_group_daily_count(group.project.id, group.id)
```

Comment:
Let's keep an eye on this and make sure it doesn't add too much load to snuba. If it does, one option we have is to only check `is_escalating` for an issue at most once per minute.

Reply:
(This might be a lot more important when we start auto archiving issues as `ignore until escalating`)